### ISSUE TITLE
doom-modeline changes not fully taking effect

### ISSUE THREAD

**Issue by shamindras** [seagle0128/doom-modeline#535]
### Thank you for the bug report

- [X] I am using the latest version of `doom-mode` related packages.
- [X] I checked [FAQ](https://github.com/seagle0128/doom-modeline#faq).
- [X] You may also try reproduce the issue using clean environment and minimal configurations with
the command `emacs -Q`.

### Bug description

@seagle0128 - thanks for creating this great package.

I've just setup `doom-modeline` and am using `straight.el` to install and set it up.
This is the code I'm using:

```lisp
;; You must run (all-the-icons-install-fonts) one time after
;; installing this package!
(use-package all-the-icons
:straight t
:if (display-graphic-p))

(use-package minions
:straight t
:hook (doom-modeline-mode . minions-mode))

(use-package doom-modeline
       :straight t
       :init
       (doom-modeline-mode 1)
       :custom
       (setq doom-modeline-buffer-file-name-style 'filename ;; Just show file name (no path)
	     doom-modeline-enable-word-count t
	     doom-modeline-buffer-encoding nil
	     doom-modeline-icon t ;; Enable/disable all icons
	     doom-modeline-modal-icon nil ;; Icon for Evil mode
	     doom-modeline-major-mode-icon t
	     doom-modeline-major-mode-color-icon t
	     doom-modeline-bar-width 3))
```

This code does display the doom-modeline as follows

But the changes made by `setq` don't seem to be taking effect. For example `doom-modeline-buffer-file-name-style 'filename` or `doom-modeline-buffer-file-name-style 'auto` does not display just the filename, but the full filepath.

Also `doom-modeline-enable-word-count t` does not display word count etc.

Importantly I can't change the width, it remains fixed and is quite large on my screen. I would like to minimize this and thought that I could vary the `doom-modeline-bar-width 3` setting.

Please see the screenshot for an example of the appearance:
<img width="1790" alt="Screen Shot 2022-06-08 at 11 30 34 PM" src="https://user-images.githubusercontent.com/7627188/172758346-2db18b7a-f209-4086-9277-b1a5c6bcd5f5.png">

In short the modeline appears but with default settings that I can't seem to vary.

### Steps to reproduce

I've tried commenting out the code above, and the `doom-modeline` does disappear. So the above code is working, though not as expected.

### Expected behavior

As per the above I expect by changing `doom-modeline-buffer-file-name-style 'filename` or `doom-modeline-buffer-file-name-style 'auto`, that the modeline should display just the filename, not the full filepath.

Similar issues with other `doom-modeline` settings.

Could anyone please explain if I'm doing anything incorrectly, or how I should change my settings to fix this issue?

### OS

MacOS

### Emacs Version

28

**Comment by seagle0128** [comment]
Please see the documentation of `use-package`. setq should be put in `:init` section, and before `(doom-modeline-mode 1). The use of the keyword `:custom` should be like this:

```elisp
:custom
  (doom-modeline-icon t)
```

**Comment by seagle0128** [comment]
Changing `doom-modeline-height` and `doom-modeline-bar-width` should work. `doom-modeline-height` respects the font height as described in docstring, thus means the mode-line height is the max of `doom-modeline-height` and font height. Generally the icons are bigger that texts, so the mode-line looks bigger than what you expected (10).

Regarding `doom-modeline-bar-width`, try setting to 0 to check the result. Please make sure using the latest.

**Comment by shamindras** [comment]
> Changing `doom-modeline-height` and `doom-modeline-bar-width` should work. `doom-modeline-height` respects the font height as described in docstring, thus means the mode-line height is the max of `doom-modeline-height` and font height. Generally the icons are bigger that texts, so the mode-line looks bigger than what you expected (10).

Thanks again @seagle0128. This is helpful. In my `init.el` I have a line as follows:

```lisp
(set-face-attribute 'default nil :font "Jetbrains Mono" :height 240)
```

I tried changing the height to to be 150, and then the modeline height does adjust when I vary the parameters.

So in order to keep the modeline small, the 'hack' that I can think of is to:

1. Use a smaller font as follows, i.e., `(set-face-attribute 'default nil :font "Jetbrains Mono" :height 240)`
2. Then the modeline is scaled down
3. Then manually zoom the emacs window until the font appears to be what I actually want, i.e., look like height 240.

Then the appearance of modeline and font is like the following, which is nice.

<img width="3190" alt="Screen Shot 2022-06-09 at 10 30 58 PM" src="https://user-images.githubusercontent.com/7627188/172978648-bb6ec190-59a1-468b-9bcd-d3384cb5210d.png">

The problem is that this zooming in is manual and tedious. Is this the only way around this modeline size issue?

**Comment by seagle0128** [comment]
@shamindras I think you need this

```elisp
(set-face-attribute 'default nil :font "Jetbrains Mono" :height 240)
(set-face-attribute 'mode-line nil :font "Jetbrains Mono" :height 150)
(set-face-attribute 'mode-line-inactive nil :font "Jetbrains Mono" :height 150)
```

**Comment by shamindras** [comment]
@seagle0128 - thanks. I tried copying the 3 lines suggested, but it did not work. That is, changing `doom-modeline-height`
and `doom-modeline-bar-width` to different parameters did not change the bar size.

For completeness, my revised config looks like this (omitting other packages):

```lisp
;; Font setup
(set-face-attribute 'default nil :font "Jetbrains Mono" :height 240)
(set-face-attribute 'mode-line nil :font "Jetbrains Mono" :height 150)
(set-face-attribute 'mode-line-inactive nil :font "Jetbrains Mono" :height 150)

;; doom-modeline setup
;; You must run (all-the-icons-install-fonts) one time after
     ;; installing this package!
     (use-package all-the-icons
       :straight t
       :if (display-graphic-p))

     (use-package minions
       :straight t
       :hook (doom-modeline-mode . minions-mode))

     (use-package doom-modeline
       :straight t
       :init
       ;; Must be set before loading doom-modeline.
       (setq doom-modeline-support-imenu t)
       (doom-modeline-mode 1)
       :hook (after-init . doom-modeline-mode)
       :custom    
       (doom-modeline-height 6)
       (doom-modeline-bar-width 2)
       (doom-modeline-icon t)
       (doom-modeline-major-mode-icon t)
       (doom-modeline-major-mode-color-icon t)
       (doom-modeline-buffer-file-name-style 'filename)
       (doom-modeline-modal-icon t) ;; Icon for Evil mode
       (doom-modeline-buffer-state-icon t)
       (doom-modeline-buffer-modification-icon t)
       (doom-modeline-minor-modes nil)
       (doom-modeline-enable-word-count nil)
       (doom-modeline-buffer-encoding t)
       (doom-modeline-indent-info nil)
       (doom-modeline-checker-simple-format t)
       (doom-modeline-vcs-max-length 12)
       (doom-modeline-env-version t))
```

However, changing _**only**_ `(set-face-attribute 'default nil :font "Jetbrains Mono" :height 150)` did do the job. But again all font sizes were set to 150 across emacs.

Is there anything I missed? Can you reproduce the above on your end?

**Comment by seagle0128** [comment]
Please try the latest. I believe changing `doom-modeline-height` and `doom-modeline-bar-width` works, unless other configurations exist.
